src/Interactive/Features/Interactive/Core/InteractiveHost.cs

```diff
@@ -231,21 +232,24 @@ private bool CheckAlive(Process process)
 
         public void Dispose()
         {
-            Dispose(joinThreads: false, disposing: true);
+            Dispose(joinThreads: true, disposing: true);
         }
 
         private void Dispose(bool joinThreads, bool disposing)
         {
-            if (disposing)
+            using (_disposeLock.DisposableWait())
             {
-                GC.SuppressFinalize(this);
-                DisposeChannel();
-            }
+                if (disposing)
+                {
+                    GC.SuppressFinalize(this);
```

Comment:
>GC.SuppressFinalize(this); [](start = 20, length = 26)

Let's move ```GC.SuppressFinalize(this)``` to the end of ```Dispose()``` to follow the common pattern.

Reply:
What pattern do you mean? I see various position of GC.SuppressFinalize(this) within Dispose(): in the beginning, in the middle and in the end. 
What is the purpose to move it to the end?

---
In reply to: [160767337](https://github.com/dotnet/roslyn/pull/24058#discussion_r160767337) [](ancestors = 160767337)

Reply:
If the object is taken out of the finalizer queue and then dispose fails, it's never gonna be finalized. So the idea is that SuppressFinalize should be the last call of Dispose in order to avoid that.

Reply:
https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/dispose-pattern